src/pybind/mgr/cephadm/http_server.py

```diff
@@ -0,0 +1,63 @@
+import cherrypy
+import threading
+import logging
+from typing import TYPE_CHECKING
+
+from cephadm.agent import AgentEndpoint
+from cephadm.service_discovery import ServiceDiscovery
+
+
+if TYPE_CHECKING:
+    from cephadm.module import CephadmOrchestrator
+
+
+def cherrypy_filter(record: logging.LogRecord) -> int:
+    blocked = [
+        'TLSV1_ALERT_DECRYPT_ERROR'
+    ]
+    msg = record.getMessage()
+    return not any([m for m in blocked if m in msg])
+
+
+logging.getLogger('cherrypy.error').addFilter(cherrypy_filter)
+cherrypy.log.access_log.propagate = False
+
+
+class CephadmHttpServer(threading.Thread):
+    def __init__(self, mgr: "CephadmOrchestrator") -> None:
+        self.mgr = mgr
+        self.agent = AgentEndpoint(mgr)
+        self.service_discovery = ServiceDiscovery(mgr)
+        self.cherrypy_shutdown_event = threading.Event()
+        super().__init__(target=self.run)
+
+    def configure_cherrypy(self) -> None:
+        cherrypy.config.update({
+            'environment': 'production',
+            'engine.autoreload.on': False,
+        })
+
+    def run(self) -> None:
+        try:
+            self.configure_cherrypy()
+            self.agent.configure()
+            self.service_discovery.configure()
```

Comment:
when there is a prometheus instance, the root_cert.pem is stored in the prometheus daemons configdir. However,  what happens when the user doesn't deploy a prometheus instance into the ceph cluster - they will still need to pick up the pem file from somewhere right? How does it work in that scenario?

Reply:
SD Root cert value is stored in the mgr store. So when no prometheus instance is deployed user can access it by using `ceph config-key dump mgr/cephadm/service_discovery_endpoint_credentials`. In the security changes I made it more straightforward so cert can be accessed directly in `mgr/cephadm/service_discovery/root/crt`.

Reply:
Perhaps add that to the doc too, so anyone wanting to set up and external prometheus isn't left wondering how they can use sd_config with the default tls that the endpoint provides?

Reply:
I added a new cmd to dump the certificate ceph orch sd dump cert so users can dump easily the current Root PEM cert and use it to configure their own monitoring stack. I updated the docs accordingly.